test/testenv/gitea.go

```diff
@@ -236,3 +237,18 @@ func UninstallGitea(ctx context.Context, input UninstallGiteaInput) {
 	_, err := removeChart.Run(nil)
 	Expect(err).ToNot(HaveOccurred())
 }
+
+func PreGiteaInstallHook(giteaInput *DeployGiteaInput, e2eConfig *clusterctl.E2EConfig) {
```

Comment:
As this only distinguishes service types, this should only return service type data, and it does not need to pass input in such case. PreGiteaInstallHook sounds ambiguous to me.

Reply:
This is done intentionally to leave space for future extensions